crazyflie/scripts/crazyflie_server.py

```diff
@@ -751,6 +757,25 @@ def _upload_trajectory_callback(self, request, response, uri="all"):
     def _start_trajectory_callback(self, request, response, uri="all"):
         self.get_logger().info("Start trajectory not yet implemented")
         return response
+    
+    def _poses_changed(self, msg):
+        """
+        Topic update callback to the motion capture lib's
+           poses topic to send through the external position
+           to the crazyflie 
+        """
+        poses = msg.poses
+        for pose in poses:
+            name = pose.name
+            x = pose.pose.position.x
+            y = pose.pose.position.y
+            z = pose.pose.position.z
+            quat = pose.pose.orientation
+            if name in self.name_dict:
+                uri = self.name_dict[name]
+                self.swarm._cfs[uri].cf.extpos.send_extpos(
+                    x, y, z, quat.x, quat.y, quat.z, quat.w)
+
```

Comment:
This would need the nan-check

Reply:
ah yes indeed, I'll implement that. So it is the mocap library that determines that we send through position only or not?.

In the cflib we do have [an option to opt out the quaternion](https://github.com/bitcraze/crazyflie-lib-python/blob/9789d1a1cc269b22ca2e35bb466a09c33be8f3c5/examples/qualisys/qualisys_hl_commander.py#L216-L219) since that is sometimes not great in not ideal setup situation, but with the active marker deck the position is still quite accurate. So that sense it's still 4+ markers but I still only want to send position through while using the QTM's result.

Perhaps I can make this a define in the python server that is on default on pose, but if we want we can put it to position, and that this part of the code checks that either quaternion is NaN or if that define is set? Or perhaps this should be handled in the motion capture library? To send position only even if the rigid bodies come in from an official mocap sdk